src/com/hms/flexyosisoftconnector/Main.java

```diff
@@ -109,10 +110,12 @@ private static void getNewPayloadToSend() {
       }
     } catch (IOException e) {
       Logger.LOG_EXCEPTION(e);
+      Logger.LOG_WARN("Exception thrown when obtaining new batch of datapoints. Will retry.");

```

Comment:
inconsistent references to data points. Here you refer to them as "datapoints" but above as "data points"